pigpen-cascading/src/main/java/pigpen/cascading/SingleIterationSeq.java

```diff
@@ -53,16 +63,24 @@ public Object first() {
   }
 
   public ISeq next() {
-    if (state.isRealized == false) {
+    if (state.rest == state) {
```

Comment:
Will there be multi-threaded access to this? Does the synchronization add any noticeable overhead?